src/server/qgsconfigcache.cpp

```diff
@@ -41,6 +44,25 @@ QgsConfigCache::QgsConfigCache()
   QObject::connect( &mFileSystemWatcher, SIGNAL( fileChanged( const QString & ) ), this, SLOT( removeChangedEntry( const QString & ) ) );
 }
 
+const QgsProject *QgsConfigCache::openProject( const QString &path )
+{
+  if ( ! mProjectCache[ path ] )
+  {
+    QgsProject *prj = new QgsProject();
+    if ( prj->read( path ) )
+    {
+      mProjectCache.insert( path, prj );
+    }
+  }
+
+  return project( path );
+}
+
+const QgsProject *QgsConfigCache::project( const QString &path ) const
+{
+  return mProjectCache[ path ];
+}
+
```

Comment:
Do you really need the `openProject` method? If this is a standard cache implementation, I would probably implement only  `project( const QString &path ) const` and handle the `openProject` logic within that method. The client code would always call `project( path )` and get the project back.